tests/test_sockets.py

```diff
@@ -315,6 +343,33 @@ def serve() -> None:
         server_sock.close()
         assert client_addr[0] == expected_client_addr
 
+    @pytest.mark.skipif(
+        sys.implementation.name == "pypy",
+        reason=(
+            "gc.get_referrers is broken on PyPy see "
+            "https://github.com/pypy/pypy/issues/5075"
+        ),
+    )
+    async def test_happy_eyeballs_refcycles(self, anyio_backend_name: str) -> None:
+        """
+        Test derived from https://github.com/python/cpython/pull/124859
+        """
+        if anyio_backend_name == "asyncio" and sys.version_info < (3, 10):
+            pytest.skip(
+                "asyncio.BaseEventLoop.create_connection creates refcycles on py 3.9"
+            )
+        ip = "127.0.0.1"
+        port = ephemeral_port_reserve.reserve(ip=ip)
```

Comment:
We have plenty of tests that need ephemeral ports. Why do we have to add a new test dependency for this specific test? (I'd be happy to add such a fixture to anyio's pytest plugin though!)

Reply:
Oh is there already a tool for this? I'll have a look

Reply:
I didn't say there was a tool yet (that's what my suggestion in parentheses was about). Rather, we currently do this in a rather manual fashion.

Reply:
done! (with #856 )